libs/NfcCoreLib/lib/Nci/phNciNfc.c

```diff
@@ -677,6 +677,77 @@ NFCSTATUS phNciNfc_Nfcee_ModeSet(void * pNciHandle,
     return wStatus;
 }
 
+NFCSTATUS phNciNfc_Nfcee_SePowerAndLinkCtrlSet(void * pNciHandle,
+                                               void * pNfceeHandle,
+                                               phNciNfc_PowerLinkModes_t eActivationMode,
+                                               pphNciNfc_IfNotificationCb_t pNotifyCb,
+                                               void *pContext)
+{
+    NFCSTATUS wStatus = NFCSTATUS_SUCCESS;
+    phNciNfc_Context_t * pNciContext = (phNciNfc_Context_t *)pNciHandle;
+    pphNciNfc_NfceeDeviceHandle_t pDevHandle =
+        (pphNciNfc_NfceeDeviceHandle_t)pNfceeHandle;
+    uint8_t *pTargetInfo;
+
+    PH_LOG_NCI_FUNC_ENTRY();
+    if (NULL == pNciContext)
+    {
+        PH_LOG_NCI_CRIT_STR("Stack not initialized");
+        wStatus = NFCSTATUS_NOT_INITIALISED;
+    }
+    else if (NULL == pNotifyCb)
+    {
+        PH_LOG_NCI_CRIT_STR("Invalid parameter passed");
+        wStatus = NFCSTATUS_INVALID_PARAMETER;
+    }
+    else
+    {
+        if ((NULL != pDevHandle) &&
+            (0 != pDevHandle->tDevInfo.bNfceeID) &&
+            (PHNCINFC_INVALID_DISCID != pDevHandle->tDevInfo.bNfceeID))
+        {
+            if (phNciNfc_IsVersion2x(pNciContext))
+            {
+                pTargetInfo = (uint8_t *)phOsalNfc_GetMemory(PHNCINFC_NFCEEMODESET_PAYLOADLEN);
+                if (NULL == pTargetInfo)
+                {
+                    PH_LOG_NCI_CRIT_STR("Memory not available(phNciNfc_Nfcee_SePowerAndLinkCtrlSet)");
```

Comment:
Here too. (Remove function name from log.)

Reply:
Done

Reply:
There is a  check for NCI version in the LibNFC  in case of NCI 1.0 can return NFCSTATUS_FEATURE_NOT_SUPPORTED,
currently it is returning NFCSTATUS_FAILED.The API being called only in case of NCI 2.0,the above same check is not required while allocating memory for payload again.

Reply:
This is true. phLibNfc_SE_PowerAndLinkControl has a Nci version check as well.
I agree that the return code NFCSTATUS_FEATURE_NOT_SUPPORTED would be a better fit.
The Nci version check can be removed as well in phNciNfc_Nfcee_SePowerAndLinkCtrlSet.
Also, looking back at the commit, I believe PHNCINFC_NFCEEMODESET_PAYLOADLEN is not the right name for this.